src/org/usfirst/frc/team2335/robot/Robot.java

```diff
@@ -1,48 +1,160 @@
+import edu.wpi.first.wpilibj.CameraServer;
 import edu.wpi.first.wpilibj.IterativeRobot;
+import edu.wpi.first.wpilibj.Relay;
 import edu.wpi.first.wpilibj.command.Command;
 import edu.wpi.first.wpilibj.command.Scheduler;
 import edu.wpi.first.wpilibj.livewindow.LiveWindow;
 import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.vision.VisionThread;
 
+/**
+ * The VM is configured to automatically run this class, and to call the
+ * functions corresponding to each mode, as described in the IterativeRobot
+ * documentation. If you change the name of this class or the package after
+ * creating this project, you must also update the manifest file in the resource
+ * directory.
+ */
+	
 public class Robot extends IterativeRobot
 {	
-	//Constants:
-	public static final double DEADZONE = 0.2; //This value is to be edited for best fir
+	//Constants:			
+	//0.2 is the value that everything before that is zeroed out
+	//So about 20% of the input will be zeroed out
+	//This created a small zone where no input will affect the output: a "deadzone"
+	public static final double DEADZONE = 0.2;
 	
 	//Axes:
 	public static final int MOVE = 1, ROTATE = 2, STRAFE = 0;
 	
+	//Buttons
+	public static final int CENTER = 2;
+	
 	//Motor ports:
 	public static final int LEFT_PORT = 0, RIGHT_PORT = 1, STRAFE_PORT = 2;
 	
+	//TODO: Remove extra newline.
+	//TODO: Remove unnecessary TODO comment
 	
 	//Subsystems:
 	public static DriveTrain driveTrain;
+	public static Vision vision;
 	public static OI oi;
 
+	//Relay
+	public static final int RELAY_PORT = 1;
+	Relay cameraLight;
+	
+	//Camera
+	//TODO: Read these off of the UsbCamera object after basic functionality.
+	//		Exclude CAMERA_ANGLE; we need to define that ourselves anyway.
+	public static final int IMG_WIDTH = 640, IMG_HEIGHT = 480;
+		
+	//Values tape
+	public static int centerX = 0, targetWidthPx = 0;
+	
+	//Camera
+	private VisionThread visionThread;
+	private Object imgLock = new Object();
+	
 	//Auto:
 	Command autonomousCommand;
-	SendableChooser chooser = new SendableChooser();
+	SendableChooser<Command> chooser = new SendableChooser<Command>();
 
 	
 	@Override
 	public void robotInit() //Runs once to initialize all global variables
 	{
+		//TODO: Possibly move the camera init to its own private function,
+		//		since it's longer and more complex.		
+		UsbCamera camera = CameraServer.getInstance().startAutomaticCapture();
+		camera.setResolution(IMG_WIDTH, IMG_HEIGHT);
+		camera.setFPS(30);
+		
+		visionThread = new VisionThread(camera, new GripPipeline(), pipeline ->
+		{
+			if(!pipeline.filterContoursOutput().isEmpty())
+			{
+				Rect r1 = Imgproc.boundingRect(pipeline.filterContoursOutput().get(0));
+								
+				synchronized (imgLock)
+				{
+					centerX = r1.x + (r1.width / 2);
+					targetWidthPx = r1.width;
+					
+					//TODO: Remove any irrelevant comments out of the cluster fuck below
```

Comment:
Remove parts of this massive comment to make it look better and easier to understand